test/weights.jl

```diff
@@ -226,3 +226,95 @@ wt = [-1, -1, -1, -1, -1]
 @test_throws ErrorException median(data, weights(wt))
 wt = [-1, -1, -1, 0, 0]
 @test_throws ErrorException median(data, weights(wt))
+
+
+
+
+# Weighted quantile tests
+data = (
+    [7, 1, 2, 4, 10],
+    [7, 1, 2, 4, 10],
+    [7, 1, 2, 4, 10, 15],
+    [1, 2, 4, 7, 10, 15],
+    [0, 10, 20, 30],
+    [1, 2, 3, 4, 5],
+    [1, 2, 3, 4, 5],
+    [30, 40, 50, 60, 35],
+    [3.7, 3.3, 3.5, 2.8],
+    [100, 125, 123, 60, 45, 56, 66],
+    [2, 2, 2, 2, 2, 2],
+    [2.3],
+    [-2, -3, 1, 2, -10],
+    [1, 2, 3, 4, 5],
+    [5, 4, 3, 2, 1],
+    [-2, 2, -1, 3, 6],
+    [-10, 1, 1, -10, -10],
+)
+wt = (
+    [1, 1/3, 1/3, 1/3, 1],
+    [1, 1, 1, 1, 1],
+    [1, 1/3, 1/3, 1/3, 1, 1],
+    [1/3, 1/3, 1/3, 1, 1, 1],
+    [30, 191, 9, 0],
+    [10, 1, 1, 1, 9],
+    [10, 1, 1, 1, 900],
+    [1, 3, 5, 4, 2],
+    [5, 5, 4, 1],
+    [30, 56, 144, 24, 55, 43, 67],
+    [0.1, 0.2, 0.3, 0.4, 0.5, 0.6],
+    [12],
+    [7, 1, 1, 1, 6],
+    [1, 0, 0, 0, 2],
+    [1, 2, 3, 4, 5],
+    [0.1, 0.2, 0.3, 0.2, 0.1],
+    [1, 1, 1, 1, 1],
+)
+quantile_answers = (    
+    [1.4444444444444444,5.0,7.999999999999999,9.0,10.0],
+    [1.0,2.0,4.0,7.0,10.0],
+    [1.9999999999999998,6.25,8.5,11.249999999999998,15.0],
+    [1.9999999999999998,6.25,8.5,11.249999999999998,15.0],
+    [4.782608695652174,12.282608695652172,19.782608695652172,27.282608695652172,30.0],
+    [1.0,1.0,4.045454545454545,5.045454545454546,5.0],
+    [4.928806133625411,5.92880613362541,6.928806133625411,7.928806133625409,5.0],
+    [35.0,40.0,50.0,53.333333333333336,60.0],
+    [3.166666666666666,3.3299999999999996,3.48,3.616666666666667,3.6999999999999993],
+    [45.89260143198091,62.7708830548926,107.84964200477326,142.34964200477327,125.0],
+    [2.0,2.0,2.0,2.0,2.0],
+    [2.3,2.3,2.3,2.3,2.3],
+    [-10.0,-9.124999999999998,-2.1875,-1.1874999999999996,2.0],
+    [1.0,4.333333333333333,5.333333333333333,6.333333333333334,5.0],
+    [1.0,1.3333333333333335,2.0,3.0,5.0],
+    [-1.5555555555555558,-0.5555555555555558,1.3333333333333324,2.666666666666666,6.0],
+    [-10.0,-10.0,-10.0,1.0,1.0],
+)
+
+
+
+
+# w = 1 corresponds to base quantile
+for i = 1:length(data)
+    for j = 1:100
+        q = rand(length(data[i]))
```

Comment:
I don't think tests on random data are really appreciated around here. Getting errors at random can be confusing, especially given that testing has been quite unstable for some time. You should probably add a call to `srand`, which people can remove to test the algorithm more extensively. But others will likely comment.